espefuse.py

```diff
@@ -836,8 +863,10 @@ def add_force_write_always(p):
         help='Run espefuse.py {command} -h for additional help')
 
     subparsers.add_parser('dump', help='Dump raw hex values of all efuses')
-    subparsers.add_parser('summary',
+    p = subparsers.add_parser('summary',
                           help='Print human-readable summary of efuse values')
+    p.add_argument('--format', help='Name of efuse register get',choices=['summary','json'],default='summary')
+    p.add_argument('--file', help='File to save the json formated summary')
```

Comment:
You can have argparse handle the file management entirely by setting `type=argparse.FileType('w')` and `default=sys.stdout`, see https://docs.python.org/2/library/argparse.html#type - that way the result will be stdout if the argument is not passed in, and the file otherwise.

Reply:
Also, nitpick but help text doesn't match bheaviour: the summary will be saved here whether JSON formatted or not.